src/core/border_router/routing_manager.hpp

```diff
@@ -589,11 +589,18 @@ class RoutingManager : public InstanceLocator
             kDeprecating,
         };
 
+        void AppendCurPrefix(Ip6::Nd::RouterAdvertMessage &aRaMessage);
+        void AppendOldPrefix(Ip6::Nd::RouterAdvertMessage &aRaMessage);
+        void Unpublish(const Ip6::Prefix &aPrefix);
+
         static void HandleTimer(Timer &aTimer);
         void        HandleTimer(void);
 
         Ip6::Prefix mPrefix;
         State       mState;
+        TimeMilli   mExpireTime;
+        Ip6::Prefix mOldPrefix;
+        TimeMilli   mOldExpireTime;
```

Comment:
It makes more sense to have a single prefix for a `LocalOnLinkPrefix` object. Should we have one more `mOldLocalOnLinkPrefix` member in `RoutingManager` rather than a `mOldPrefix` in `LocalOnLinkPrefix`?

Reply:
I treat `LocalOnLinkprefix` as the entity deciding/managing on-link prefixes. May make sense if we can rename it later to `OnLinkPrefixManager`? (can do this in next PR when we extend it to be an array of deprecating prefixes).